### PR TITLE
[docs] Update ES output config

### PR DESCRIPTION
**Pull request by bmorelli25**
Updates the Elasticsearch output config with APM Server fields. Unfortunately, I can't come up with a way to do this without introducing a bunch of new conditionals.

I've swapped out `fields.log_type` with `processor.event`, but open to other ideas, or using regexp if there are better examples.

**APM Issue:** closes https://github.com/elastic/apm-server/issues/2130. 
**Beats PR:** https://github.com/elastic/beats/pull/12034

### CODE REVIEW

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 1]
```diff
@@ -265,6 +266,26 @@ With this configuration, all events with `log_type: normal` are sent to an
...
+ifdef::apm-server[]
+You can set the index dynamically by using a format string to access any event
+field. For example, this configuration uses the field, `processor.event`,
+to set the index:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["http://localhost:9200"]
+  index: "apm-%\{[observer.version]\}-%\{[processor.event]\}-%\{+yyyy.MM.dd}\" <1>
```
Comment (jalvz) [inline docs/copied-from-beats/outputconfig.asciidoc]: i'd note `observer` is the apm-server, for people not familiar with ECS

**Comment by jalvz** [comment]
> swapped out fields.log_type with processor.event

great idea :100:

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 2]
```diff
@@ -340,6 +362,70 @@ This configuration results in indices named `sev1`, `sev2`, and `sev3`.
...
+      when.contains:
+        processor.event: "span"
+  
+   - index: "apm-%{[observer.version]}-metric-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "metric"
+  
+   - index: "apm-%{[observer.version]}-onboarding-%{+yyyy.MM.dd}"
+      when.contains:
+        processor.event: "onboarding"
```
Comment (simitt) [inline docs/copied-from-beats/outputconfig.asciidoc]: I was planning on simplifying the default config a bit (within the ILM PR). As you described in this PR it is now also possible to use `processor.event` in the regex, leading to `index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"`.  With this we can simplify the handling without any changes in the outcome. 
```
output.elasticsearch:
  hosts: ["http://localhost:9200"]
  index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"
  indices:
   - index: "apm-%{[observer.version]}-sourcemap"
      when.contains:
        processor.event: "sourcemap"
```

But as this change is _only in my head_ so far, what you have looks great!

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 3]
```diff
@@ -340,6 +362,70 @@ This configuration results in indices named `sev1`, `sev2`, and `sev3`.
...
+  indices:
+    - index: "%{[processor.event]}"
+      mappings:
+        sourcemap:    "apm-sourcemap"
+        error:        "apm-error"
+        transaction:  "apm-transaction"
+        span:         "apm-span"
+        metric:       "apm-metric"
+        onboarding:   "apm-onboarding"
+      default:        "apm"
```
Comment (simitt) [inline docs/copied-from-beats/outputconfig.asciidoc]: neat! Wasn't aware of that.

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: Woah, does it work in that order? I thought APM Server went with the first match. I was playing around with basically the same idea yesterday and got it to work in this order:
```
output.elasticsearch:
  hosts: ["http://localhost:9200"]
  indices:
   - index: "apm-%{[observer.version]}-sourcemap"
      when.contains:
        processor.event: "sourcemap"
   - index: "apm-%{[observer.version]}-%{[processor.event]}-%{+yyyy.MM.dd}"
```

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: This is a good point. I'll note that in the `<1>` callout

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: super neat!

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: If you're planning on simplifying the default config within the ILM PR, let's definitely get on the same page. I'd love to update this documentation with what you're thinking.

**Comment by bmorelli25** [comment]
Merged https://github.com/elastic/beats/pull/12034 to persist these changes and ran `make update-beats-docs`. There are a few new asciidoctor `eval` -> `if` changes that don't impact APM Server documentation.

**Comment by bmorelli25** [comment]
@simitt, I guess it probably makes more sense for me to merge this as-is for now. Once it's ILM time, let's return to this conversation: https://github.com/elastic/apm-server/pull/2161#discussion_r280327007!

**Pull request merged**